PennMobile/HomeViewController.swift

```diff
@@ -0,0 +1,312 @@
+        } else {
+            return 60.0
+        }
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        //adding table view cell programmatically
+        let setting = customSettings[indexPath.section]
+        
+        let cell = tableView.dequeueReusableCell(withIdentifier: setting, for: indexPath)
+        
+        if let cell = cell as? WeatherCell {
+            cell.delegate = self
+            cell.reloadData()
+            return cell
+        } else if let cell = cell as? AgendaCell {
+            cell.delegate = self
+            cell.reloadData()
+            return cell
+        } else if let cell = cell as? ReservationCell {
+            cell.delegate = self
+            cell.reloadData()
+            return cell
+        } else if let cell = cell as? DiningCell {
+            cell.delegate = self
+            cell.reloadData()
+            return cell
+        }
+        return cell
+    }
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return customSettings.count
+    }
+    
+    // Set the spacing between sections
+    override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        if section == 0 {
+            return 0
+        } else if customSettings[section - 1] == "Weather" {
+            return 20
+        }
+        return cellSpacingHeight
+    }
+    
+    // Make the background color show through
+    override func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let headerView = UIView()
+        headerView.backgroundColor = UIColor.clear
+        return headerView
+    }
+    
+    func handleShowSettings() {
+        let svc = SettingsViewController()
+        svc.delegate = self
+        navigationController?.pushViewController(svc, animated: false)
+    }
+    
+    func refreshAllData(_ refreshControl: UIRefreshControl) {
+        NetworkManager.getWeatherData(callback: { (dictionary) in
+            
+            if let temp = dictionary["temp"] as? NSNumber, let description = dictionary["description"] as? String {
+                self.weather = Weather(temperature: temp.stringValue, description: description)
+            }
+            
+            //TODO: Implement other getters
+            
+            NetworkManager.getAnnouncementData(callback: { (announcements) in
+                
+                //TODO: temporary
+                self.agendaAnnouncement = announcements.first?.title
+                
+                for announcement in announcements {
+                    let today = Date.currentLocalDate
+                    if announcement.start > today && announcement.end > today {
+                        //TODO: ask Tiff
+                        
```

Comment:
No it's not. Home Page shouldn't have been included in this pull request. Sincerest apologies.